File: ros_code/ros_c++/S_node.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>


void callback(std_msgs::String m)
{
    printf( m.data.c_str() );
    printf("\n");


}
int main(int argc, char* argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc,argv,"S_node");
    ros::NodeHandle n;
    ros::Subscriber sub =n.subscribe("topic",10,callback);
    while(ros::ok())
    {
        ros::spin();
    }
    return 0;
}

File: ros_py/S_node.py
#!/usr/bin/env python3
#coding = utf-8



import rospy
from std_msgs.msg import String

def callback(m):
  rospy.loginfo(m.data)


if __name__ == "__main__":
  rospy.init_node("s_node")
  rospy.logwarn("node successfully init")
  sub=rospy.Subscriber("topic_py",String,callback=callback,queue_size=10)
  while not rospy.is_shutdown() :
    rospy.spin()